### PR TITLE
Several mro related fixes/enhancements for 2.x

### LINKED ISSUE #213: MRO: sonarqube branch should be set to '*' in case of multirepobuild = true
**Describe the bug**
as the MRO creates branches dynamically - and relies on a source code review report to generated docs .. the shared lib should set `sonarQubeBranch` to `*` automatically.

https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/OdsPipeline.groovy#L133

@metmajer @martsec - fyi. (current workaround .. set to `sonarQubeBranch` to `*` on master before assemble)

### LINKED ISSUE #214: startOpenShiftBuild should retry to get the build on 2.x
**Describe the bug**
In some cases (Openshift bug?) - although the build is finished (--follow) the build returns still running

image

https://github.com/opendevstack/ods-jenkins-shared-library/blob/2.x/vars/stageStartOpenshiftBuild.groovy#L35

we should retry to get the build a couple times ... to make sure this is mega resilient.

### PR DESCRIPTION
**Pull request by clemensutschig**
closes #214 
closes #213 

for 2.x

### CODE REVIEW

File: src/org/ods/OdsPipeline.groovy [hunk 1]
```diff
@@ -125,6 +128,7 @@ class OdsPipeline implements Serializable {
     context.displayNameUpdateEnabled = false
     context.ciSkipEnabled = false
     context.notifyNotGreen = false
+    context.sonarQubeBranch = '*'
```
Comment (michaelsauter) [inline src/org/ods/OdsPipeline.groovy:131]: Doesn't this need a setter?

Comment (clemensutschig) [reply src/org/ods/OdsPipeline.groovy:131]: @michaelsauter  - is in the below commits :)

File: src/org/ods/OdsContext.groovy [hunk 2]
```diff
@@ -312,6 +312,10 @@ class OdsContext implements Context {
       config.sonarQubeBranch
   }
 
+  void setSonarQubeBranch(String sonarQubeBranch) {
```
Comment (michaelsauter) [inline src/org/ods/OdsContext.groovy:315]: We also need this in the interface, `Context.groovy`.

Comment (clemensutschig) [reply src/org/ods/OdsContext.groovy:315]: @michaelsauter  - worked without ... reason?

Comment (clemensutschig) [reply src/org/ods/OdsContext.groovy:315]: fixed

**Pull request merged**